### PR TITLE
More exam updates

### PR DESCRIPTION
**Pull request by christianmemije**
### Summary

…

### Reviewer guidance

…

### References

…

----

### Contributor Checklist

- [ ] Contributor has fully tested the PR manually
- [ ] PR has the correct target branch and milestone
- [ ] PR has 'needs review' or 'work-in-progress' label
- [ ] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label

### Reviewer Checklist

- [ ] Automated test coverage is satisfactory
- [ ] Reviewer has fully tested the PR manually
- [ ] PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- [ ] External dependencies files were updated (`yarn` and `pip`)
- [ ] Documentation is updated
- [ ] Link to diff of internal dependency change is included
- [ ] CHANGELOG.rst is updated for high-level changes
- [ ] Contributor is in AUTHORS.rst

### CODE REVIEW

File: kolibri/plugins/coach/assets/src/views/exams/exam-report-page/index.vue [hunk 1]
```diff
@@ -33,14 +33,10 @@
         :key="i"
       >
         <h3>
-          {{ viewByGroups ? reportGrouping[0].group.name : $tr('allLearners') }}
+          {{ viewByGroups ? reportGrouping[0].group.name || $tr('ungrouped') : $tr('allLearners') }}
```
Comment (jonboiser) [inline kolibri/plugins/coach/assets/src/views/exams/exam-report-page/index.vue]: Appears to cause a problem if `reportGrouping[0]` is undefined, which can happen if you have a Learner Group with zero members.

File: kolibri/plugins/coach/assets/src/views/exams/exam-report-page/index.vue [hunk 2]
```diff
@@ -148,6 +145,12 @@
           reportGroupings = this.learnerGroups.map(group =>
             this.examTakers.filter(learner => learner.group.id === group.id)
           );
+          reportGroupings = orderBy(
+            reportGroupings,
+            [grouping => grouping[0].group.name.toUpperCase()],
```
Comment (jonboiser) [inline kolibri/plugins/coach/assets/src/views/exams/exam-report-page/index.vue]: Appears to cause a problem if `grouping[0]` is undefined, which can happen if you have a Learner Group with zero members.

File: kolibri/plugins/coach/assets/src/views/exams/exam-report-page/index.vue (hunk 1 above)
Comment (jonboiser) [inline kolibri/plugins/coach/assets/src/views/exams/exam-report-page/index.vue:36]: Based on comment below, `reportGrouping[0]` can be undefined under certain circumstances.

File: kolibri/plugins/learn/assets/src/state/actions/main.js [hunk 3]
```diff
@@ -405,6 +405,13 @@ export function calcQuestionsAnswered(attemptLogs) {
   return questionsAnswered;
 }
 
+function _canViewExam(active, closed) {
```
Comment (jonboiser) [inline kolibri/plugins/learn/assets/src/state/actions/main.js]: i would probably just pass a single argument `exam` or `examLog` into these helpers. It's more consistent with how these functions are implemented in `AssignedExamCards`

File: kolibri/plugins/style_guide/assets/src/views/shell/component-docs.vue [hunk 4]
```diff
@@ -40,7 +40,7 @@
         </tr>
         <tr v-for="(event, i) in api.events" :key="i">
           <td>{{ event.name }}</td>
-          <td>{{ event.description ? event.description : '-' }}</td>
+          <td>{{ event.description ? event.description : '–' }}</td>
```
Comment (jonboiser) [inline kolibri/plugins/style_guide/assets/src/views/shell/component-docs.vue]: could condense to `event.description || '–' `

Comment (jonboiser) [reply kolibri/plugins/style_guide/assets/src/views/shell/component-docs.vue]: same with the other `<td>`s

File: kolibri/plugins/learn/assets/src/state/actions/main.js [hunk 5]
```diff
@@ -471,28 +480,33 @@ export function showExam(store, examId, questionNumber) {
         // Local copy of exam attempt logs
         const attemptLogs = {};
 
-        if (userId) {
```
Comment (christianmemije) [inline kolibri/plugins/learn/assets/src/state/actions/main.js:474]: This if statement was redundant.

File: kolibri/plugins/learn/assets/src/state/actions/main.js [hunk 6]
```diff
@@ -651,7 +665,10 @@ export function setAndSaveCurrentExamAttemptLog(store, contentId, itemId, curren
         // Add this attempt log to the Collection for future caching.
         examAttemptLogCollection.set(examAttemptLogModel);
         resolve();
-      })
+      }),
+    () => {
+      this.$router.replace({ name: ClassesPageNames.CLASS_ASSIGNMENTS });
```
Comment (christianmemije) [inline kolibri/plugins/learn/assets/src/state/actions/main.js:670]: @rtibbles Thoughts on this?

**Pull request merged**